docs/posts/flink-quickstart/index.qmd

```diff
@@ -0,0 +1,262 @@
+        "provinceId": "int32",
+    }
+)
+
+source_configs = {
+    "connector": "kafka",
+    "topic": "payment_msg",
+    "properties.bootstrap.servers": "localhost:9092",
+    "properties.group.id": "test_3",
+    "scan.startup.mode": "earliest-offset",
+    "format": "json",
+}
+
+t = con.create_table(
+    "payment_msg",
+    schema=source_schema,
+    tbl_properties=source_configs,
+    watermark=ibis.watermark(
+        time_col="createTime", allowed_delay=ibis.interval(seconds=15)
+    ),
+)
+
+# create sink Table
+sink_schema = ibis.schema(
+    {
+        "province_id": "int32",
+        "pay_amount": "float64",
+    }
+)
+
+sink_configs = {
+    "connector": "kafka",
+    "topic": "sink",
+    "properties.bootstrap.servers": "localhost:9092",
+    "format": "json",
+}
+
+con.create_table(
+    "total_amount_by_province_id", schema=sink_schema, tbl_properties=sink_configs
+)
+```
+
+#### Perform calculations
+
+Compute the total pay amount per province in the past 10 seconds (as of each message, for the province in the incoming message).
+
+```{python}
+agged = t.select(
+    province_id=t.provinceId,
+    pay_amount=t.payAmount.sum().over(
+        range=(-ibis.interval(seconds=10), 0),
+        group_by=t.provinceId,
+        order_by=t.createTime,
+    ),
+)
+```
+
+Finally, emit the query result to the sink table.
+
+```{python}
+con.insert("total_amount_by_province_id", agged)
+```
+
+### Remote execution
+
+You can also submit the example to the [remote cluster started using Docker Compose](#spinning-up-the-services-using-docker-compose). The `window_aggregation.py` file in the cloned repository contains the [same steps that we performed for local execution](#local-execution). We will [use the method described in the official Flink documentation](https://nightlies.apache.org/flink/flink-docs-release-1.18/docs/deployment/cli/#submitting-pyflink-jobs).
+
+::: {.callout-tip}
+You can find the `./bin/flink` executable with the following command:
+
+```bash
+python -c'from pathlib import Path; import pyflink; print(Path(pyflink.__spec__.origin).parent / "bin" / "flink")'
+```
+:::
+
+My full command looks like this:
+
+```bash
+/opt/miniconda3/envs/ibis-dev/lib/python3.10/site-packages/pyflink/bin/flink run --jobmanager localhost:8081 --python window_aggregation.py
+```
+
+The command will exit after displaying a submission message:
+
+```bash
+Job has been submitted with JobID b816faaf5ef9126ea5b9b6a37012cf56
+```
+
+## Viewing the results
+
+Similar to how we viewed messages in the `payment_msg` topic, we can print results from the `sink` topic:
+
+```{python}
+consumer = KafkaConsumer("sink")
+for msg in islice(consumer, 10):
+    print(msg)
+```
+
+Voilà! You've run your first streaming application using Ibis.
+
+## Shutting down the Compose environment
+
+Press <kbd>Ctrl</kbd>+<kbd>C</kbd> to stop the Docker Compose containers. Once stopped, run `docker compose down` to remove the services created for this tutorial.
```

Comment:
Does this work? Typically containers don't respond to SIGINT from the caller unless explicitly enabled to do so.

Not a blocker in any event.

Reply:
Works for me. :)

Reply:
Looks something like this:

```bash
ibis-flink-example-init-kafka-1      | Created topic sink.
ibis-flink-example-init-kafka-1      | Successfully created the following topics:
ibis-flink-example-init-kafka-1      | payment_msg
ibis-flink-example-init-kafka-1      | sink
ibis-flink-example-init-kafka-1 exited with code 0
ibis-flink-example-data-generator-1  | Connected to Kafka
ibis-flink-example-data-generator-1  | Producing 20000 records to Kafka topic payment_msg
kafka                                | [2024-01-10 16:17:08,314] INFO [BrokerToControllerChannelManager id=1 name=forwarding] Node 1 disconnected. (org.apache.kafka.clients.NetworkClient)
^CGracefully stopping... (press Ctrl+C again to force)
Aborting on container exit...
[+] Stopping 5/5
 ✔ Container ibis-flink-example-taskmanager-1     Stopped                                                                                                                                                  1.6s 
 ✔ Container ibis-flink-example-data-generator-1  Stopped                                                                                                                                                 10.3s 
 ✔ Container ibis-flink-example-jobmanager-1      Stopped                                                                                                                                                  1.3s 
 ✔ Container ibis-flink-example-init-kafka-1      Stopped                                                                                                                                                  0.0s 
 ✔ Container kafka                                Stopped                                                                                                                                                  1.6s 
canceled
(base) deepyaman@Deepyamans-MacBook-Air ibis-flink-example % 
```